src/app/modules/MainScreen/MainScreen.tsx

```diff
@@ -14,7 +14,8 @@ type MainScreenProps = WrappedComponentProps &
 
 class MainScreen extends PureComponent<MainScreenProps> {
   handleToSecondScreen = () => {
-    this.props.navigation.navigate(Modules.ProfileScreenModule.name);
+    const { navigate } = this.props.navigation;
```

Comment:
personally, I prefered previous version as you're not reusing `navigate` property anywhere